python/mxnet/foo/nn/legacy.py

```diff
@@ -0,0 +1,138 @@
+                self._reg_params[param] = \
+                    self.params.get(param, grad_req='write', shape=shape, init=init)
+
+        aux_params = symbol.list_auxiliary_states()
+        aux_params = tuple(map(eliminate_prefix, tuple(aux_params)))
+
+        for aux_param, shape in zip(aux_params, aux_shapes):
+            if aux_param not in self._variables:
+                init = init_dict.get(aux_param)
+                self._reg_params[aux_param] = \
+                    self.params.get(aux_param, grad_req='null', shape=shape, init=init)
+
+    def _alias(self):
+        return self._operator_name.lower()
+
+    def ndarray_forward(self, *args, **kwargs):
+        """ Imperative call.
+
+        Parameters
+        ----------
+        args : `NDArray` inputs. The order of `args` should be identical to that of `variable`,
+            which is specified by calling `__init__`.
+        kwargs : Please do not provide any key-word argument.
+            `kwargs` is only for implementation purpose.
+        """
+        kwargs.update(dict(zip(self._variables, args)))
+        kwargs.update(self._kwargs)
+        return self._operator(**kwargs)
+
+    def symbolic_forward(self, *args, **kwargs):
+        """ Symbolic call.
+
+        Parameters
+        ----------
+        args : `Symbol` inputs. The order of `args` should be identical to that of `variable`,
+            which is specified by calling `__init__`.
+        kwargs : Please do not provide any key-word argument.
+            `kwargs` is only for implementation purpose.
+        """
+        kwargs.update(dict(zip(self._variables, args)))
+        kwargs.update(self._kwargs)
+        return getattr(_sym, self._operator_name)(name=self._prefix, **kwargs)
+
+    def generic_forward(self, F, x, *args, **kwargs):
+        """ Not implemented. """
+        raise NotImplementedError()
+
+
+# pylint: disable=locally-disabled, invalid-name
+Activation = lambda **kwargs: _LegacyLayer('Activation', **kwargs)
+BatchNorm = lambda **kwargs: _LegacyLayer('BatchNorm', **kwargs)
+Convolution = lambda **kwargs: _LegacyLayer('Convolution', **kwargs)
+Deconvolution = lambda **kwargs: _LegacyLayer('Deconvolution', **kwargs)
+Dropout = lambda **kwargs: _LegacyLayer('Dropout', **kwargs)
+Embedding = lambda **kwargs: _LegacyLayer('Embedding', **kwargs)
+FullyConnected = lambda **kwargs: _LegacyLayer('FullyConnected', **kwargs)
+LeakyReLU = lambda **kwargs: _LegacyLayer('LeakyReLU', **kwargs)
+Pooling = lambda **kwargs: _LegacyLayer('Pooling', **kwargs)
+RNN = lambda **kwargs: _LegacyLayer('RNN', **kwargs)
+
+# pylint: disable=locally-disabled, invalid-name
+ReLU = lambda: _LegacyLayer('Activation', act_type='relu')
+Sigmoid = lambda: _LegacyLayer('Activation', act_type='sigmoid')
+Tanh = lambda: _LegacyLayer('Activation', act_type='tanh')
+
+# pylint: disable=locally-disabled, invalid-name
+RNNReLU = lambda **kwargs: _LegacyLayer('RNN', mode='rnn_relu', **kwargs)
```

Comment:
legacy interface should be the same with functional interface. RNN is enough. We don't need these